test/runtime/kafka.go

```diff
@@ -0,0 +1,147 @@
+	var disallowTopic string = "disallowTopic"
+	var MaxMessages int = 5
+
+	initialize := func() {
+		if initialized == true {
+			return
+		}
+		logger = log.WithFields(log.Fields{"testName": "RuntimeKafka"})
+		logger.Info("Starting")
+		docker, cilium = helpers.CreateNewRuntimeHelper("runtime", logger)
+		docker.NetworkCreate(helpers.CiliumDockerNetwork, "")
+		initialized = true
+	}
+
+	containers := func(mode string) {
+
+		images := map[string]string{
+			"zook":   "digitalwonderland/zookeeper",
+			"client": "cilium/kafkaclient2",
+		}
+
+		switch mode {
+		case "create":
+			for k, v := range images {
+				docker.ContainerCreate(k, v, helpers.CiliumDockerNetwork, fmt.Sprintf("-l id.%s", k))
+			}
+			zook, err := docker.ContainerInspectNet("zook")
+			Expect(err).Should(BeNil())
+
+			docker.ContainerCreate("kafka", "wurstmeister/kafka", helpers.CiliumDockerNetwork, fmt.Sprintf(
+				"-l id.kafka -e KAFKA_ZOOKEEPER_CONNECT=%s:2181 ", zook["IPv4"]))
+
+		case "delete":
+			for k := range images {
+				docker.ContainerRm(k)
+			}
+			docker.ContainerRm("kafka")
+		}
+	}
+
+	createTopic := func(name string) {
+		logger.Infof("Creating new kafka topic %s", name)
+		res := docker.ContainerExec("client", fmt.Sprintf(
+			"/opt/kafka/bin/kafka-topics.sh --create --zookeeper zook:2181 "+
+				"--replication-factor 1 --partitions 1 --topic %s", name))
+		res.ExpectSuccess()
+	}
+
+	consumer := func(topic string, maxMsg int) string {
+		return fmt.Sprintf(
+			"docker exec client /opt/kafka/bin/kafka-console-consumer.sh --bootstrap-server "+
+				"kafka:9092 --topic %s --max-messages %d --timeout-ms 30000", topic, maxMsg)
+	}
+
+	producer := func(topic string, message string) {
+		cmd := fmt.Sprintf(
+			"echo %s | docker exec -i client /opt/kafka/bin/kafka-console-producer.sh "+
+				"--broker-list kafka:9092 --topic %s",
+			message, topic)
+		docker.Node.Exec(cmd)
+	}
+
+	BeforeEach(func() {
+		initialize()
+		containers("create")
+		epsReady := cilium.WaitEndpointsReady()
+		Expect(epsReady).Should(BeTrue())
+	})
+
+	AfterEach(func() {
+		if CurrentGinkgoTestDescription().Failed {
+			cilium.ReportFailed()
+		}
+		cilium.Exec("policy delete --all")
+		containers("delete")
+	})
+
+	It("Kafka Policy Ingress", func() {
+		_, err := cilium.PolicyImport(cilium.GetFullPath("Policies-kafka.json"), 300)
+		Expect(err).Should(BeNil())
+
+		endPoints, err := cilium.PolicyEndpointsSummary()
+		Expect(err).Should(BeNil())
+		Expect(endPoints[helpers.Enabled]).To(Equal(1))
+		Expect(endPoints[helpers.Disabled]).To(Equal(2))
+
+		createTopic(allowedTopic)
+		createTopic(disallowTopic)
+
+		res := docker.ContainerExec("client",
+			"/opt/kafka/bin/kafka-topics.sh --list --zookeeper zook:2181")
+		res.ExpectSuccess("Cannot get kafka topics")
+
+		By("Allowed topic")
+		ctx, cancel := context.WithCancel(context.Background())
+		data := cilium.Node.ExecContext(ctx, consumer(allowedTopic, MaxMessages))
+
+		//TODO: wait until ready
+		helpers.Sleep(5)
```

Comment:
We can fix this sleep by creating consumer in a specified group: `bin/kafka-console-consumer.sh --bootstrap-server localhost:9092 --topic allowedTopic --max-messages 5 --timeout-ms 30000 --group cilium-test-consumer`

and later wait until `kafka-consumer-groups.sh` doesn't have "has no active members" in stderr. In bash it would be done like that:
`until ! bin/kafka-consumer-groups.sh --describe --group cilium-test-consumer  --bootstrap-server localhost:9092 2>&1 >/dev/null | grep "has no active members"; do sleep 1; done`.

Above can be done as improvement in follow-up PR, as this was absent in bash test.